components/automate-ui/src/app/pages/project/details/project-details.component.ts

```diff
@@ -85,28 +95,25 @@ export class ProjectDetailsComponent implements OnInit, OnDestroy {
           (getRulesSt !== EntityStatus.loadingSuccess);
       });
 
-    this.store.select(projectFromRoute).pipe(
-      filter(identity),
-      takeUntil(this.isDestroyed)
-      ).subscribe((state) => {
-        this.project = { ...state };
-        this.store.dispatch(new GetRulesForProject({ project_id: this.project.id }));
-        this.store.select(allRules).subscribe((rules) => {
-          this.rules = rules;
-        });
+    combineLatest([
+      this.store.select(getStatus),
+      this.store.select(getAllRulesForProjectStatus),
+      this.store.select(projectFromRoute),
+      this.store.select(allRules)
+    ]).pipe(
+        filter(([getProjectSt, getRulesSt, _projectState, _allRulesState]) =>
+          getProjectSt === EntityStatus.loadingSuccess &&
+          getRulesSt === EntityStatus.loadingSuccess),
```

Comment:
Wondering if this could be split into having a `loading$` Observable in its own `combineLatest` as Lance did in his recent PR, then you could leverage the new `allLoadedSuccessfully` function he is introducing. See https://github.com/chef/automate/pull/2151.
I bring that up because I think it would then make this a bit simpler.

Reply:
I added a `allLoaded` function but it wasn't obvious to me how to easily move `isLoading` to an observable since it's being used in typescript functions. Can make a follow up PR if this is easy to fix.